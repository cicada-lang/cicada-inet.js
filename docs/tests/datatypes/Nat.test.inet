require "Nat.inet"

import Nat, zero, zero as z, add1, add from "Nat.inet"