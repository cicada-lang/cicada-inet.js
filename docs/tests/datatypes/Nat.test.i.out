(zero₁₆)-value!target-(add₆)
(zero₁₅)-value addend-(add₆)
(add₆)-return

(zero₁₅)-value

(add1₁₈)-value!target-(add₇)
(add1₁₇)-value addend-(add₇)
(zero₁₈)-value prev-(add1₁₈)
(zero₁₇)-value prev-(add1₁₇)
(add₇)-return

(add1₁₇)-value prev-(add1₁₉)
(zero₁₇)-value prev-(add1₁₇)
(add1₁₉)-value

(add1₂₃)-value!target-(add₁₀)
(add1₂₁)-value!target-(add₉)
(add₉)-return addend-(add₁₁)
(add₁₀)-return target-(add₁₁)
(add1₂₂)-value addend-(add₁₀)
(zero₂₂)-value prev-(add1₂₃)
(zero₂₁)-value prev-(add1₂₂)
(add1₂₀)-value addend-(add₉)
(zero₂₀)-value prev-(add1₂₁)
(zero₁₉)-value prev-(add1₂₀)
(add₁₁)-return

(add1₂₀)-value prev-(add1₂₇)
(add1₂₇)-value prev-(add1₂₆)
(add1₂₆)-value prev-(add1₂₅)
(zero₁₉)-value prev-(add1₂₀)
(add1₂₅)-value

(add1₂₈)-value!target-(add₁₆)
(zero₂₃)-value addend-(add₁₆)
(zero₂₄)-value prev-(add1₂₈)
(add₁₆)-return

(zero₂₃)-value prev-(add1₂₉)
(add1₂₉)-value

(add1₃₀)-value prev-(add1₃₅)
(add1₃₅)-value prev-(add1₃₄)
(zero₂₅)-value prev-(add1₃₀)
(add1₃₄)-value

(add1₃₇)-value!target-(add₂₃)
(add₂₃)-return target-(nat_erase₆)
(add1₃₆)-value addend-(add₂₃)
(zero₂₉)-value prev-(add1₃₇)
(zero₂₈)-value prev-(add1₃₆)
(zero₃₀)-value

(zero₃₀)-value

(add1₄₀)-value!target-(add₂₅)
(add₂₅)-return target-(nat_dup₆)
(add1₃₉)-value addend-(add₂₅)
(zero₃₂)-value prev-(add1₄₀)
(zero₃₁)-value prev-(add1₃₉)
(nat_dup₆)-second
(nat_dup₆)-first

(zero₃₄)-value prev-(add1₄₅)
(zero₃₃)-value prev-(add1₄₄)
(add1₄₅)-value prev-(add1₄₃)
(add1₄₄)-value prev-(add1₄₂)
(add1₄₂)-value
(add1₄₃)-value

(add1₄₉)-value!target-(add₂₈)
(add1₄₇)-value!target-(add₂₇)
(add₂₇)-return mulend-(mul₅)
(add₂₈)-return target-(mul₅)
(add1₄₈)-value addend-(add₂₈)
(zero₃₈)-value prev-(add1₄₉)
(zero₃₇)-value prev-(add1₄₈)
(add1₄₆)-value addend-(add₂₇)
(zero₃₆)-value prev-(add1₄₇)
(zero₃₅)-value prev-(add1₄₆)
(mul₅)-return

(zero₄₃)-value prev-(add1₆₃)
(add1₅₇)-value prev-(add1₆₀)
(add1₆₃)-value prev-(add1₅₇)
(add1₆₀)-value prev-(add1₅₄)
(add1₅₄)-value

(add1₆₉)-value!target-(add₄₂)
(add1₆₈)-value!target-(add₄₁)
(add1₆₆)-value!target-(add₄₀)
(add1₆₅)-value!target-(add₃₉)
(add₄₀)-return mulend-(mul₁₀)
(add₄₂)-return target-(mul₁₀)
(add₄₁)-return addend-(add₄₂)
(zero₅₃)-value prev-(add1₆₉)
(add1₆₇)-value addend-(add₄₁)
(zero₅₂)-value prev-(add1₆₈)
(zero₅₁)-value prev-(add1₆₇)
(add₃₉)-return addend-(add₄₀)
(zero₅₀)-value prev-(add1₆₆)
(add1₆₄)-value addend-(add₃₉)
(zero₄₉)-value prev-(add1₆₅)
(zero₄₈)-value prev-(add1₆₄)
(mul₁₀)-return

(zero₆₀)-value prev-(add1₁₀₀)
(add1₉₁)-value prev-(add1₉₇)
(add1₇₈)-value prev-(add1₉₄)
(add1₁₀₀)-value prev-(add1₉₁)
(add1₉₇)-value prev-(add1₈₈)
(add1₉₄)-value prev-(add1₈₅)
(add1₈₈)-value prev-(add1₈₁)
(add1₈₅)-value prev-(add1₇₅)
(add1₈₁)-value prev-(add1₇₈)
(add1₇₅)-value

(add1₁₀₁)-value prev-(add1₁₀₄)
(zero₆₈)-value prev-(add1₁₀₁)
(add1₁₀₄)-value

(add1₁₁₀)-value prev-(add1₁₀₉)
(zero₇₁)-value prev-(add1₁₁₀)
(add1₁₀₉)-value

(add1₁₁₁)-value prev-(add1₁₂₀)
(add1₁₂₀)-value prev-(add1₁₁₈)
(zero₇₃)-value prev-(add1₁₁₁)
(add1₁₁₈)-value

